microprofile.jdt/org.eclipse.lsp4mp.jdt.test/src/main/java/org/eclipse/lsp4mp/jdt/core/faulttolerance/java/MicroProfileFaultToleranceJavaDiagnosticsTest.java

```diff
@@ -67,27 +67,53 @@ public void asynchronousNonFutureOrCompletionStage() throws Exception {
 		diagnosticsParams.setUris(Arrays.asList(javaFile.getLocation().toFile().toURI().toString()));
 		diagnosticsParams.setDocumentFormat(DocumentFormat.Markdown);
 
-		Diagnostic d1 = d(34, 11, 17, 
-				"The annotated method does not return an object of type Future or CompletionStage", 
-				DiagnosticSeverity.Error,
-				MicroProfileFaultToleranceConstants.DIAGNOSTIC_SOURCE,
+		Diagnostic d1 = d(34, 11, 17,
+				"The annotated method does not return an object of type Future, CompletionStage or Uni",
+				DiagnosticSeverity.Error, MicroProfileFaultToleranceConstants.DIAGNOSTIC_SOURCE,
 				MicroProfileFaultToleranceErrorCode.FAULT_TOLERANCE_DEFINITION_EXCEPTION);
-		
-		Diagnostic d2 = d(39, 11, 15, 
-				"The annotated method does not return an object of type Future or CompletionStage", 
-				DiagnosticSeverity.Error,
-				MicroProfileFaultToleranceConstants.DIAGNOSTIC_SOURCE,
+
+		Diagnostic d2 = d(39, 11, 15,
+				"The annotated method does not return an object of type Future or CompletionStage or Uni",
+				DiagnosticSeverity.Error, MicroProfileFaultToleranceConstants.DIAGNOSTIC_SOURCE,
 				MicroProfileFaultToleranceErrorCode.FAULT_TOLERANCE_DEFINITION_EXCEPTION);
-		
-		Diagnostic d3 = d(44, 11, 36, 
-				"The annotated method does not return an object of type Future or CompletionStage", 
-				DiagnosticSeverity.Error,
-				MicroProfileFaultToleranceConstants.DIAGNOSTIC_SOURCE,
+
+		Diagnostic d3 = d(44, 11, 36,
+				"The annotated method does not return an object of type Future or CompletionStage or Uni",
+				DiagnosticSeverity.Error, MicroProfileFaultToleranceConstants.DIAGNOSTIC_SOURCE,
+				MicroProfileFaultToleranceErrorCode.FAULT_TOLERANCE_DEFINITION_EXCEPTION);
+
+		assertJavaDiagnostics(diagnosticsParams, utils, d1, d2, d3);
+	}
+
+	@Test
+	public void asynchronousClassNonFutureOrCompletionStage() throws Exception {
+		IJavaProject javaProject = loadMavenProject(MicroProfileMavenProjectName.microprofile_fault_tolerance);
+		IJDTUtils utils = JDT_UTILS;
+
+		MicroProfileJavaDiagnosticsParams diagnosticsParams = new MicroProfileJavaDiagnosticsParams();
+		IFile javaFile = javaProject.getProject()
+				.getFile(new Path("src/main/java/org/acme/AsynchronousFaultToleranceClassResource.java"));
+		diagnosticsParams.setUris(Arrays.asList(javaFile.getLocation().toFile().toURI().toString()));
+		diagnosticsParams.setDocumentFormat(DocumentFormat.Markdown);
+
+		Diagnostic d1 = d(32, 11, 17,
+				"The annotated method does not return an object of type Future or CompletionStage or Uni",

```

Comment:
This should fail right ? There's a comma right after Future in the original text.

Reply:
Ah yes sorry

Reply:
Please make sure you run the tests locally, particularly if modifying test cases and functionality like this.

Reply:
Forgot a save